### PR TITLE
Pass tdepth as needed for hillslope model

### LINKED ISSUE #20: MOSART_SIMYR in config_component.xml not setup correctly
MOSART_SIMYR in config_component.xml isn't setup correctly. The default value uses command line syntax, "-simyr 2000", while the specifc settings us a fixed year.

It looks like as follows...

``` xml
  <entry id="MOSART_SIMYR">
    <type>char</type>
    <valid_values></valid_values>
    <default_value>-simyr 2000</default_value>
    <values>
      <value compset="1850.*_MOSART">1850</value>
      <value compset="HIST.*_MOSART">1850</value>
      <value compset="PIPD.*_MOSART">1850</value>
      <value compset="2000.*_MOSART">2000</value>
    </values>
    <group>run_component_mosart</group>
    <file>env_run.xml</file>
    <desc>MOSART build-namelist options</desc>
  </entry>
```

In practice it isn't actually used in the code and hence doesn't matter right now. So it doesn't actually do anything in buildnml nor do the namelist defaults do anything based on it. The intention for it is to be able to set the initial conditions to either present day or pre-industrial. But, since we aren't providing initial conditions it doesn't really matter.

### LINKED ISSUE #36: Pass water depth states to LND
For the CTSM hillslope model we need to pass two new state fields from MOSART to LND (River tributaries water depth, and River tributaries bankfull (maximum) water depth) [Sr_tdepth and Sr_tdepth_max]. These are the first state fields for MOSART to LND, so that also means there needs to be mapping files for states from ROF to LND.

The cime issue that relates to this is: https://github.com/ESMCI/cime/issues/3699

@swensosc

### LINKED ISSUE #44: Remove references to Flrl_rofdto
@mvertens points out that Flrl_rofdto is no longer used, so it should be removed. This should be a simple cleanup, which can probably be done without any significant testing.

### LINKED ISSUE #45: frivinp_rtm is still required even if MOSART is turned off
Even when MOSART_MODE==NULL the buildnml still requires frivinp_rtm to be set. It would be good to turn this requirement off when MOSART_MODE is NULL.

### LINKED ISSUE #47: do_rtm=FALSE is not working with NUOPC
The mosartOff testmod seems to be failing when run with NUOPC.

I'm testing this what will be mosart1_0_44, with future ctsm5.1.dev062, but I think this is a problem with the previous version as well.

The test

SMS_Ld1_D_Vnuopc.f10_f10_mg37.I2000Clm50BgcCru.cheyenne_intel.mosart-mosartOff

fails.

### PR DESCRIPTION
**Pull request by ekluzek**
Pass tdepth/tdepth_max as needed for the hilllslope model. Add a trigger for it from CMEPS.

Also don't set direction file if MOSART is turned off. Remove do_rtm namelist trigger and use MOSART_MODE
instead.

Fixes #47
Fixes #36
Fixes #45
Fixes #44
Fixes #20

### CODE REVIEW

**Comment by ekluzek** [comment]
@billsacks I'd like you to look at this and let's talk about it this afternoon. The specific change I'm not convinced is right is that I have buildnml change MOSART_MODE under certain conditions. This does work. But, philosophically I think it likely should be done in buildlib rather than buildnml. So I wanted to hear what you thought about that.

File: cime_config/buildnml [hunk 1]
```diff
@@ -40,8 +40,12 @@ def _create_namelists(case, confdir, inst_string, infile, nmlgen, data_list_path
     #----------------------------------------------------
     config = {}
     config['mosart_mode'] = case.get_value("MOSART_MODE")
-    config['mosart_flood_mode'] = case.get_value("MOSART_FLOOD_MODE")
     config['clm_accel'] = case.get_value("CLM_ACCELERATED_SPINUP")
+    if ( config['clm_accel'] != "off" ):
+       config['mosart_mode'] = "NULL"
+       case.set_value( "MOSART_MODE", config['mosart_mode'] )
```
Comment (ekluzek) [inline cime_config/buildnml:46]: I probably should do this in buildlib rather than here.

**Comment by billsacks** [comment]
You might not like my answer:

I don't like having buildnml _or_ buildlib change xml values. I feel like this is confusing from both a user and developer perspective. I think it's better to maintain a consistent flow: The compset can set the default value of xml variables, and xml variables can set the default value of namelist variables; other than that, it should be up to the user to set things explicitly. I don't feel like this needs to be followed 100%, but there should be a very good reason for not following it.

In this case, I think I understand the reason for setting MOSART_MODE automatically, so there is one less thing for a user to set when running accelerated spinup. However, in my view, that benefit does not justify violating what I feel should be a general principle in the user interface of CESM's Case Control System.

I can see a few alternatives for how to handle this:

(1) Introduce either compsets or a user mods directory for turning on CLM spinup. A compset would be straightforward here (use `SROF`), but I'm thinking a user mods directory might be best to avoid a proliferation of compsets. Users can then add the appropriate `--user-mods-dir` setting when creating a case where they want spinup active. This would set `CLM_ACCELERATED_SPINUP` along with setting `ROF_MODE` appropriately (note generic rename: see point (3) for more on this, but it could be renamed even if keeping this variable defined in MOSART: then you don't need separate logic all over the place for `MOSART_MODE` vs. `RTM_MODE`, etc.).

(2) If (1) isn't possible, or you want to support manually changing `CLM_ACCELERATED_SPINUP` after a case is created, then: Instead of setting `MOSART_MODE` (which again, I'd suggest renaming to `ROF_MODE`) based on `CLM_ACCELERATED_SPINUP`, instead leave this up to the user, but issue an error message if they haven't done this. This logic should probably be in CTSM, since it is really CTSM-specific. I'd suggest something like: If `CLM_ACCELERATED_SPINUP` is on and `ROF_MODE` is "ACTIVE", issue an error saying that the user should also set `ROF_MODE` to "NULL". However, also provide an option of `ROF_MODE` = "ACTIVE_FORCE" that would bypass this check. So the standard value of `ROF_MODE` would be "ACTIVE", and then if someone turns on `CLM_ACCELERATED_SPINUP`, then they should either set `ROF_MODE` to "NULL" or to "ACTIVE_FORCE" depending on what they want. I (significantly) prefer the explicitness of this solution to the current invisible change of `MOSART_MODE`.

And a tangential point, alluded to above:

(3) Based on where `MOSART_MODE` is referenced – in CMEPS and not (as far as I can see) in MOSART itself: it really feels to me like there should be a generically-named `ROF_MODE` variable defined in CMEPS rather than having every runoff model define its own version of this variable that then requires additional changes in CMEPS. The individual runoff models can then remain blissfully unaware of this variable, and any logic related to the variable can be in CMEPS (or in CLM for checking compatibility with `CLM_ACCELERATED_SPINUP` as noted in point (2)).

**Comment by ekluzek** [comment]
OK, I'll make a new issue regarding this. I didn't like my solution either, which is why I wanted you to look at it. In the short term we'll bring this in like it is, so that I can do the next CTSM tag. But, we'll decide on a longer term solution and implement that when we are able to.

**Pull request merged**